regexp/escape.ts

```diff
@@ -93,5 +93,5 @@ export function escape(str: string): string {
   return str.replaceAll(
     RX_REGEXP_ESCAPE,
     (m) => RESERVED_CHARS[m as keyof typeof RESERVED_CHARS],
-  );
+  ).replace(/^\d/, (m) => `\\x${m.codePointAt(0)!.toString(16)}`);
```

Comment:
```suggestion
  ).replace(/^[0-9a-zA-Z]/, (m) => `\\x${m.codePointAt(0)!.toString(16)}`);
```

https://tc39.es/proposal-regex-escaping/#sec-regexp.escape behavior and rationale:

> a. If escaped is the empty String and c is matched by either [DecimalDigit](https://tc39.es/ecma262/#prod-DecimalDigit) or [AsciiLetter](https://tc39.es/ecma262/#prod-AsciiLetter), then
>       i. NOTE: Escaping a leading digit ensures that output corresponds with pattern text which may be used after a \0 character escape or a [DecimalEscape](https://tc39.es/ecma262/#prod-DecimalEscape) such as \1 and still match S rather than be interpreted as an extension of the preceding escape sequence. Escaping a leading ASCII letter does the same for the context after \c.

Reply:
Good catch! Thanks